java/vector/src/main/java/org/apache/arrow/vector/complex/ListViewVector.java

```diff
@@ -0,0 +1,945 @@
+      final int prevOffset = offsetBuffer.getInt((index - 1) * OFFSET_WIDTH);
+      final int prevSize = sizeBuffer.getInt((index - 1) * SIZE_WIDTH);
+      final int currOffSet = prevOffset + prevSize;
+      offsetBuffer.setInt(index * OFFSET_WIDTH, currOffSet);
+      sizeBuffer.setInt(index * SIZE_WIDTH, 0);
+    }
+
+    BitVectorHelper.unsetBit(validityBuffer, index);
+    lastSet = index;
+  }
+
+  /**
+   * Start new value in the ListView vector.
+   * There are a few cases that are handled in this function.
+   * There are two main scenarios that need to be considered.
+   * The first scenario is simple insertion where indices are continuously updated.
+   * The other scenario is the event of non-continuous writing,
+   * the offset buffer needs to be updated.
+   *
+   * @param index index of the value to start
+   * @return offset of the new value
+   */
+  @Override
+  public int startNewValue(int index) {
+    while (index >= getValidityAndSizeValueCapacity()) {
+      reallocValidityAndSizeAndOffsetBuffers();
+    }
+
+    if (lastSet >= index) {
+      lastSet = index - 1;
+    }
+
+    if (index == 0) {
+      offsetBuffer.setInt(0, 0);
+    } else if (index > lastSet) {
+      /* when skipping indices, we need to update the offset buffer */
+      /* setting offset from lastSet + 1 to index (included) */
+      for (int i = lastSet + 1; i <= index; i++) {
+        if (i == 0) {
+          offsetBuffer.setInt(0, 0);
+          continue;
+        }
+        final int prevOffSet = offsetBuffer.getInt((i - 1L) * OFFSET_WIDTH);
+        final int prevSize = sizeBuffer.getInt((i - 1L) * SIZE_WIDTH);
+        final int currOffSet = prevOffSet + prevSize;
+        offsetBuffer.setInt(i * OFFSET_WIDTH, currOffSet);
+      }
+    } else {
+      final int prevOffset = offsetBuffer.getInt((index - 1) * OFFSET_WIDTH);
+      final int prevSize = sizeBuffer.getInt((index - 1) * SIZE_WIDTH);
+      final int currOffSet = prevOffset + prevSize;
+      offsetBuffer.setInt(index * OFFSET_WIDTH, currOffSet);
+    }
+
+    BitVectorHelper.setBit(validityBuffer, index);
+    lastSet = index;
+    return offsetBuffer.getInt(index * OFFSET_WIDTH);
+  }
+
+  private int getLengthOfChildVector() {
+    int length = 0;
+    for (int i = 0; i < valueCount; i++) {
+      length += sizeBuffer.getInt(i * SIZE_WIDTH);
+    }
+    return length;
+  }
+
+  /**
+   * Constructing a ListViewVector when the offsets, sizes and field vector are available.
+   * <p>
+   * Steps taken follow the workflow used in creating a ListViewVector with the API
+   * used in ListVector.
+   *
+   * @param offSetBuffer new offSet buffer to be set
+   * @param sizeBuffer new size buffer to be set
+   * @param validityBuffer new validity buffer to be set
+   * @param elementFieldVec new elements to be appended to the field vector
+   * @param valueCount number of lists to be set
+   */
+  public void set(ArrowBuf offSetBuffer, ArrowBuf sizeBuffer, ArrowBuf validityBuffer,
```

Comment:
I'd rather we leave this out

Reply:
Yes of course, I am going to remove this entirely. And update things with the proposed new API, I just need to update the test cases to follow that API and then I will remove this.

Reply:
Removed.